src/Codec/JVM/Attr.hs

```diff
@@ -35,6 +35,7 @@ data Attr
     , codeAttrs :: [Attr] }
   | AStackMapTable [(Offset, StackMapFrame)]
   | AInnerClasses InnerClassMap
+  | ASignature Text
```

Comment:
@puffnfresh Since `Attr` is a shared data type between decoding & encoding, I think it should be something that is useful on both the parsing side as we as generating side. So I would suggest that you and @Abhiroop come to a consensus on the representation. I think that would also remove the need for   the `addFormals` function you've defined.